File: 프로그래머스/연습문제/Lv1/달리기 경주/달리기 경주.py
def solution(players, callings):
    m = {}
    for i in range(len(players)):
        m[players[i]] = i

    for call in callings:
        idx = m[call]

        players[idx] = players[idx-1]
        players[idx-1] = call

        m[call] = idx-1
        m[players[idx]] = idx

    return players

File: 프로그래머스/연습문제/Lv1/달리기 경주/달리기 경주.cpp
#include <string>
#include <vector>
#include <map>
using namespace std;

vector<string> solution(vector<string> players, vector<string> callings) {
    map<string, int> m; // <이름, 인덱스>
    for(int i=0; i<players.size(); i++)
        m[players[i]] = i;
    
    for(string call : callings){
        int idx = m[call];
        
        players[idx] = players[idx-1];
        players[idx-1] = call;
        
        m[call] = idx-1;
        m[players[idx]] = idx;
    }
    
    return players;
}
